src/Microsoft.Net.Http.Headers/NameValueHeaderValue.cs

```diff
@@ -390,13 +415,109 @@ private static void CheckValueFormat(StringSegment value)
             // Either value is null/empty or a valid token/quoted string
             if (!(StringSegment.IsNullOrEmpty(value) || (GetValueLength(value, 0) == value.Length)))
             {
-                throw new FormatException(string.Format(System.Globalization.CultureInfo.InvariantCulture, "The header value is invalid: '{0}'", value));
+                throw new FormatException(string.Format(CultureInfo.InvariantCulture, "The header value is invalid: '{0}'", value));
             }
         }
 
         private static NameValueHeaderValue CreateNameValue()
         {
             return new NameValueHeaderValue();
         }
+
+        // See https://tools.ietf.org/html/rfc7230#section-3.2.6
+        private static StringSegment UnescapeAsQuotedString(StringSegment input)
```

Comment:
Or did you leave it here because you didn't want to make assumptions about the input? Some assumptions are ok so long as they are well documented.